imagededup/utils/plotter.py

```diff
@@ -0,0 +1,130 @@
+import matplotlib.gridspec as gridspec
+import matplotlib.pyplot as plt
+from matplotlib import figure
+from pathlib import Path, PosixPath
+from typing import Dict, Union, List
+
+import numpy as np
+from PIL import Image
+
+
+def _formatter(val: Union[int, np.float32]):
+    """
+    For printing floats only upto 3rd precision. Ints are unchanged.
+    """
+    if isinstance(val, np.float32):
+        return f'{val:.3f}'
+    else:
+        return val
+
+
+def _plot_images(
+    image_dir: PosixPath,
+    orig: str,
+    image_list: List,
+    scores: bool = False,
+    outfile: str = None,
+) -> None:
+    """
+    Plotting function for plot_duplicates() defined below.
+
+    Args:
+        image_dir: image directory where all files in duplicate_map are present.
+        orig: filename for which duplicates are to be plotted.
+        image_list: List of duplicate filenames, could also be with scores (filename, score).
+        scores: Whether only filenames are present in the image_list or scores as well.
+        outfile:  Name of the file to save the plot.
+    """
+    n_ims = len(image_list)
+    ncols = 4  # fixed for a consistent layout
+    nrows = int(np.ceil(n_ims / ncols)) + 1
+    fig = figure.Figure(figsize=(10, 14))
+
+    gs = gridspec.GridSpec(nrows=nrows, ncols=ncols)
+    ax = plt.subplot(
+        gs[0, 1:3]
+    )  # Always plot the original image in the middle of top row
+    ax.imshow(Image.open(image_dir / orig))
+    ax.set_title('Original Image: {}'.format(orig))
+    ax.axis('off')
+
+    for i in range(0, n_ims):
+        row_num = (i // ncols) + 1
+        col_num = i % ncols
+
+        ax = plt.subplot(gs[row_num, col_num])
+        if scores:
+            ax.imshow(Image.open(image_dir / image_list[i][0]))
+            val = _formatter(image_list[i][1])
+            title = ' '.join([image_list[i][0], f'({val})'])
+        else:
+            ax.imshow(Image.open(image_dir / image_list[i]))
+            title = image_list[i]
+
+        ax.set_title(title, fontsize=6)
+        ax.axis('off')
+    gs.tight_layout(fig)
+
+    if outfile:
+        plt.savefig(outfile)
+
+    plt.show()
+
+
+def _validate_args(
+    image_dir: Union[PosixPath, str], duplicate_map: Dict, filename: str
+) -> None:
+    """Argument validator for plot_duplicates() defined below."""
+
+    image_dir = Path(image_dir)
+    assert (
+        image_dir.is_dir()
+    ), 'Provided image directory does not exist! Please provide the image directory where all files are present!'
+
+    if not isinstance(duplicate_map, dict):
+        raise ValueError('Please provide a valid Duplicate map!')
+    if filename not in duplicate_map.keys():
+        raise ValueError(
+            'Please provide a valid filename present as a key in the duplicate_map!'
+        )
+
+
+def plot_duplicates(
```

Comment:
I get this error when running plot_duplicates

![image](https://user-images.githubusercontent.com/19587525/65524110-916d8500-deed-11e9-9cd0-f028fe4a1418.png)

Reply:
Fixed. Also added a test case to check that nonposixpath image directories are accepted.